content-services/content-sources-frontend
# Fixes 4743: add delete and bulk delete of snapshots

## Summary
This PR adds the UI changes needed for manual snapshot deletion. In the snapshot list modal a checkbox and a kebab menu has been added for each snapshot which can be used to bulk delete or single delete the chosen snapshots.
New "delete snapshots confirmation" modal has been added where the to be deleted snapshots are displayed together with any templates that use those snapshots, so that the users are warned this change will also affect those templates. A separate warning alert is displayed in that case.

Related ticket: [[HMS-4743](https://issues.redhat.com/browse/HMS-4743)]

## Testing steps
- Workflow checks:
  - Create a repo with at least 3 snaphshots. _(you can use the [zoo fakerepos](https://fedorapeople.org/groups/katello/fakerepos/) and just change numbers when editing the repo)_
  - Create at least 2 templates using the custom repo, where one is `use_latest` and the other has a date in the past.
  - Verify that the snapshot(s) and the templates using them are present in the delete snapshots modal, or not present in the case of the "middle" snapshot.
  - Delete a single snapshot and verify the repo, template and snapshots content gets updated.
  - Reset the repo's snapshots, and bulk delete multiple snapshots and verify the same is true.

- General checks:
  - Verify you can't delete all the snapshots for a repository.
  - Verify you can't delete snapshots without `repoWrite` permissions.
  - Verify you can't delete snapshots of RH repositories.
